tests/wlr_foreign_toplevel_management_v1.cpp

```diff
@@ -895,21 +885,15 @@ TEST_F(ForeignToplevelHandleTest, can_activate_foreign)
 
 TEST_F(ForeignToplevelHandleTest, can_close_foreign)
 {
-    bool should_close{false};
-    xdg_toplevel.add_close_notification(
-        [&should_close]()
-        {
-            should_close = true;
-        });
+    EXPECT_CALL(xdg_toplevel, close()).Times(0);
```

Comment:
I'm not *sure* that this does what you want? GMock expectations are additive; how does this `EXPECT_CALL().Times(0)` interact with the previous `EXPECT_CALL().Times(AnyNumber())`?

Reply:
I believe new more specific `EXPECT_CALL`'s supersede old more general ones. If you remove the `EXPECT_CALL(xdg_toplevel, close()).Times(1);` a few lines down the test fails.